src/submission/submission-service.ts

```diff
@@ -230,19 +234,20 @@ export namespace operations {
         `No active submission found with programId: ${programId} & versionId: ${versionId}`,
       );
     }
-    const newActiveSubmission = _.cloneDeep(exsistingActiveSubmission) as ActiveClinicalSubmission;
     if (
       exsistingActiveSubmission.state === SUBMISSION_STATE.VALID ||
```

Comment:
shouldn't this if statement really just be `if (existingActiveSubmission.state !== SUBMISSION_STATE.OPEN)`?

Reply:
thtat doesn't take into account SUBMISSION_STATE.INVALID

Reply:
If I read this correctly, this is checking that validation has already been run so we can return the current activeSubmission instead of re-running validation. If that is correct, then when this is INVALID you can just return that, no need to validate again?